keps/sig-apps/3140-TimeZone-support-in-CronJob/README.md

```diff
@@ -0,0 +1,504 @@
+a [request was raised to support setting time zones](https://github.com/kubernetes/kubernetes/issues/47202).
+The initial [response from SIG-Apps and SIG-Architecture](https://github.com/kubernetes/kubernetes/issues/47202#issuecomment-360820586)
+at the time was that introducing such functionality would require cluster operator
+to manually include TimeZone database since golang did not have one.
+Starting from [golang 1.15](https://go.dev/doc/go1.15) `time/tzdata` package can
+be embedded in the binary thus removing the requirement for external database.
+At that time the majority of the focus was towards [moving CronJob to GA](https://github.com/kubernetes/enhancements/issues/19)
+so the effort to support TimeZone was again delayed. Now that we have CronJob
+fully GA-ed it's time to satisfy the original request.
+
+### Goals
+
+- Add the field `.spec.timeZone` which allows specifying a valid TimeZone
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Add the field `.spec.timeZone` to the CronJob resource. The cronjob controller
+will take the field into account when scheduling the next Job run. In case the
+field is not specified or is empty the controller will maitain the current
+behavior which is to rely on the time zone of the kube-controller-manager
+process.
+
+### Notes/Constraints/Caveats (Optional)
+
+The current mechanism, which will still be the default behavior heavily relies
+on the time zone of the kube-controller-manager process which is hard for a
+regular user to figure out. Exposing an explicit field for setting a time zone
+will allow CronJob authors to simplify the user experience when creating CronJobs.
+
+### Risks and Mitigations
+
+- Outdated time zone in the golang might lead to wrong schedule times
+
+This problem can be mitigated with a fresh build of kube-controller-manager with
+updated golang version, but it heavily relies on the go community to keep the
+time zone database up-to-date.
+
+- Malicious user can create multiple CronJobs with different time zone which can
+actually trigger Jobs at the exact same time
+
+Cluster administrators should enforce quota to ensure not that many Jobs and
+CronJobs can be created per user.
+
+
+## Design Details
+
+### CronJob API
+
+The `CronJobSpec` structure is expanded with new `TimeZone` field which allows
+specifying the name of the time zone to be used. Missing or empty value of the
+field indicates the current behavior, which relies on the time zone of the
+kube-controller-manager process.
+
+```golang
+
+type CronJobSpec struct {
+
+	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	Schedule string
+
+  // Time zone for the above schedule
+  TimeZone *string
+
+}
+```
+
+The value provided in `TimeZone` field will be validated against the embedded
+golang timezone database.
```

Comment:
Does this mean that changes to time zone databases need to be applied to the API server and kube-controller-manager in a particular order?

Reply:
No need for manual action, just follow the general guideline for updating the cluster: https://kubernetes.io/docs/tasks/administer-cluster/cluster-upgrade/